core/domain/user_services.py

```diff
@@ -656,3 +656,38 @@ def get_user_dashboard_stats(user_id):
             'total_plays': 0,
             'average_ratings': None
         }
+
+def get_weekly_dashboard_stats(user_id):
```

Comment:
Actually, thinking about this more, is it actually necessary to save one list element for each user every week? This could quickly grow very big. Do you actually need the info from past weeks? From what I see, I think you only need the current and the previous weeks' info stored.